sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -63,8 +63,10 @@ private String generateBinaryQuery(@NonNull Criteria criteria, @NonNull List<Pai
         final String sqlKeyword = criteria.getType().getSqlKeyword();
         parameters.add(Pair.of(parameter, subjectValue));
 
-        if (CriteriaType.isFunction(criteria.getType())) {
-            return getFunctionCondition(ignoreCase, sqlKeyword, subject, parameter);
+        if (CriteriaType.isFunction(criteria.getType()) && ! CriteriaType.isUnary(criteria.getType())) {
```

Comment:
```suggestion
        if (CriteriaType.isFunction(criteria.getType()) && !CriteriaType.isUnary(criteria.getType())) {
```

Reply:
I don't quite get this - StartsWith and EndsWith allow setting the boolean parameters whether to do case sensitive search or not - So I assume you would want to pass true into takesCaseSensitiveParam - but isUnary(STARTS_WITH) would return false - meaning you would end-up setting to takesCaseSensitiveParam  and end-up calling UPPER again (being a horribly inefficient query)